httphandler.go

```diff
@@ -41,65 +33,78 @@ func (h HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	switch r.Method {
 	case "GET":
-		h.get(id, w)
+		h.get(sid, w)
 	case "HEAD":
-		h.head(id, w)
+		h.head(sid, w)
 	case "PUT":
-		h.put(id, w, r)
+		h.put(sid, w, r)
 	default:
 		w.WriteHeader(http.StatusMethodNotAllowed)
 		w.Write([]byte("only GET is supported"))
 	}
 }
 
-func (h HTTPHandler) get(id ChunkID, w http.ResponseWriter) {
-	b, err := h.s.GetChunk(id)
-	switch err.(type) {
-	case nil:
-		w.WriteHeader(http.StatusOK)
-		w.Write(b)
-	case ChunkMissing:
-		w.WriteHeader(http.StatusNotFound)
-		fmt.Fprintf(w, "chunk %s not found", id)
-	default:
-		w.WriteHeader(http.StatusInternalServerError)
-		msg := fmt.Sprintf("failed to retrieve chunk %s:%s", id, err)
-		fmt.Fprintln(w, msg)
-		fmt.Fprintln(os.Stderr, msg)
+func (h HTTPHandler) parseChunkId(sid string, w http.ResponseWriter) (ChunkID, error) {
+	// Parse the ID and verify the format
+	cid, err := ChunkIDFromString(sid)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("invalid chunk id"))
+		return ChunkID{}, err
 	}
+	return cid, err
 }
 
-func (h HTTPHandler) head(id ChunkID, w http.ResponseWriter) {
-	if h.s.HasChunk(id) {
+func (h HTTPHandler) get(sid string, w http.ResponseWriter) {
+	cid, err := h.parseChunkId(sid, w)
+	if err != nil {
+		return
```

Comment:
The parseChunkId method itself is writing the error. I return the error to get, head and put so that I can return early. I did this because those two lines were repeated and I thought I might as well encapsulate them. Perhaps the method name is not intuitive though?

Reply:
Yes, you're right, I didn't see that. As for naming, I'll leave that to you. Don't really have a strong opinion on it.